mapknowledge/__init__.py

```diff
@@ -184,34 +184,34 @@ def __init__(self, store_directory=None,
             sckan_build = self.__scicrunch.build()
             if sckan_build is not None:
                 self.__sckan_provenance['sckan'] = {
-                    'date': sckan_build['released']
+                    'scicrunch':{
+                        'url': scicrunch_api,
+                        'date': sckan_build['released']
+                    }
                 }
             if log_build:
                 scicrunch_build = (f" built at {sckan_build['released']}" if sckan_build is not None else '')
                 release_version = 'production' if scicrunch_release == SCICRUNCH_PRODUCTION else 'staging'
                 log.info(f"With {release_version} SCKAN{scicrunch_build} from {self.__scicrunch.sparc_api_endpoint}")
         else:
             self.__scicrunch = None
-            log.info('Without SCKAN')
+            log.info('Without Scicrunch')
         if npo:
             # self.__npo_db = NpoSparql()
             self.__npo_db = Npo(npo_release)
             self.__npo_entities = set(self.__npo_db.connectivity_paths().keys())
             self.__npo_entities.update(self.__npo_db.connectivity_models().keys())
             npo_builds = self.__npo_db.build()
             if len(npo_builds):
-                self.__sckan_provenance['npo'] = {
-                    'date': npo_builds['released']
-                }
-                self.__sckan_provenance['apinatomy'] = {
-                    'date': npo_builds['date'],
-                    'path': npo_builds['path'],
-                    'sha': npo_builds['sha']
+                if 'sckan' not in self.__sckan_provenance: self.__sckan_provenance['sckan'] = {}
+                self.__sckan_provenance['sckan']['npo'] = {
```

Comment:
Why `['sckan']['npo']` and not `['npo']`??

Reply:
This is actually something I'm quite doubtful about because previously mapmaker combined Scicrunch and NPO, and this functionality is still there. Because now, mapmaker can just use NPO or Scicrunch, so the sckan_provenance information will be different based on sources used:

- Scicrunch and NPO:
'sckan': {
            'npo': {
                   'date': '2024-03-26',
                   'path': 'https://github.com/SciCrunch/NIF-Ontology/tree/sckan-2024-03-26',
                   'release': 'sckan-2024-03-26',
                   'sha': 'd5b7a991e75f22c3d3e5239e6ba0ad95fed40f1d'
            },
           'scicrunch': {
                   'date': '2024-03-26',
                   'url': 'https://scicrunch.org/api/1'
           }
}

- Scicrunch only:
'sckan': {
            'scicrunch': {
                   'date': '2024-03-26',
                   'url': 'https://scicrunch.org/api/1'
           }
}

- NPO only:
'sckan': {
            'npo': {
                   'date': '2024-03-26',
                   'path': 'https://github.com/SciCrunch/NIF-Ontology/tree/sckan-2024-03-26',
                   'release': 'sckan-2024-03-26',
                   'sha': 'd5b7a991e75f22c3d3e5239e6ba0ad95fed40f1d'
                      }
}

Should this use the previous results? `sckan` is for Scicrunch source and `NPO` is for github source, because I think `Scicrunch` and `NPO` are also `sckan`?

Reply:
But why the top level `sckan` key? Can't `sckan_provenance` just have two possible entries, one for `scicrunch` and one for `npo` (and historically, one for `apinatomy`)??

I see `SCKAN` as the generic term for K-Core's connectivity knowledge, which originally was only available from an endpoint on SciCrunch. Now we are distinguishing between the version of SciCrunch and that of the NPO knowledge (which we are getting directly from GitHub).

Reply:
Yes, using `scicrunch` and `npo` as entries will be clearer. For this, I have push a commit, removing `sckan` at top level.